streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/StreamStreamJoinNode.java

```diff
@@ -98,12 +108,19 @@ public void writeToTopology(final InternalTopologyBuilder topologyBuilder) {
         topologyBuilder.addStateStore(thisWindowStoreBuilder, thisWindowedStreamProcessorName, otherProcessorName);
         topologyBuilder.addStateStore(otherWindowStoreBuilder, otherWindowedStreamProcessorName, thisProcessorName);
 
-        final StreamsConfig streamsConfig = topologyBuilder.getStreamsConfig();
-        if (streamsConfig == null || StreamsConfig.InternalConfig.getBoolean(streamsConfig.originals(), ENABLE_KSTREAMS_OUTER_JOIN_SPURIOUS_RESULTS_FIX, true)) {
+        if (props == null || StreamsConfig.InternalConfig.getBoolean(toMap(props), ENABLE_KSTREAMS_OUTER_JOIN_SPURIOUS_RESULTS_FIX, true)) {
```

Comment:
Instead of using `toMap`, can't we just to `new HashMap<String,String>(props)`

Reply:
HashMap does not have a constructor that accepts Properties.

Reply:
This works (note that `Properties implements Map<Object, Object>)`:
```
Properties p = new Properties();
Map<String, Object> foo = new HashMap(p);
```

Reply:
Done. It wasn't working initially, I probably had a syntax error.